### PR TITLE
Verify PVC by writing a file

### PR DESCRIPTION
**Pull request by brybacki**
**What this PR does / why we need it**:

Adds a verification steps to make sure the PVC is restored
and not re-imported. Also delete VM in foreground, so it waits
for all dependent objects to also disappear.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/resource_filtering_test.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"fmt"
 	kubernetes "k8s.io/client-go/kubernetes"
+	//"k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/api/errors"
```
Comment (alromeros) [inline tests/resource_filtering_test.go:7]: Should we leave this commented?

File: tests/utils.go [hunk 2]
```diff
@@ -361,7 +361,10 @@ func DeleteDataVolume(kvClient kubecli.KubevirtClient, namespace, name string) e
 
 func DeleteVirtualMachine(client kubecli.KubevirtClient, namespace, name string) error {
 	return wait.PollImmediate(pollInterval, waitTime, func() (bool, error) {
-		err := client.VirtualMachine(namespace).Delete(name, &metav1.DeleteOptions{})
+		propagationForeground := metav1.DeletePropagationForeground
+		err := client.VirtualMachine(namespace).Delete(name, &metav1.DeleteOptions{
+			PropagationPolicy: &propagationForeground,
```
Comment (alromeros) [inline tests/utils.go:366]: Just out of curiosity, why do we want this? Isn't this the default behavior?

**Review by alromeros — COMMENTED** [review]
Looks good! I like how clean the pod creation/verification looks.

File: tests/utils.go (hunk 2 above)
Comment (brybacki) [reply tests/utils.go:366]: Good question, I need to find out what is the default behavior

Comment (brybacki) [reply tests/utils.go:366]: I am sure I have seen tests failed during restore, let me rerun test to check it in an experiment

Comment (brybacki) [reply tests/utils.go:366]: when I remove this, one test fails, when trying to restore, and the PVC was still not removed
https://github.com/brybacki/kubevirt-velero-plugin/blob/dfa59f7e5cdde18b8bd419c8557abcee970600f3/tests/resource_filtering_test.go#L661

Comment (brybacki) [reply tests/utils.go:366]: I suppose more tests can fail, if not checking for deletion of all involved resources

File: tests/resource_filtering_test.go (hunk 1 above)
Comment (brybacki) [reply tests/resource_filtering_test.go]: removed

**Comment by brybacki** [comment]
@alromeros wdyt?

File: tests/utils.go (hunk 2 above)
Comment (alromeros) [reply tests/utils.go:366]: Thanks for the clarification!

**Pull request merged**